homeassistant/components/luxtronik/config_flow.py

```diff
@@ -0,0 +1,203 @@
+"""Config flow to configure the Luxtronik heatpump controller integration."""
+# region Imports
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.dhcp import DhcpServiceInfo
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+
+from .const import CONF_HA_SENSOR_PREFIX, DEFAULT_HOST, DEFAULT_PORT, DOMAIN, LOGGER
+from .coordinator import LuxtronikCoordinator
+from .lux_helper import discover
+
+# endregion Imports
+
+
+PORT_SELECTOR = vol.All(
```

Comment:
Set the `NumberSelector` directly in `STEP_USER_DATA_SCHEMA` and then convert to int in validation or when reading the value